File: Leetcode-Problems/Minimum Absolute Difference.cpp
class Solution
{
public:
    vector<vector<int>> minimumAbsDifference(vector<int> &arr)
    {
        vector<vector<int>> ans;
        int diff = INT_MAX;
        sort(arr.begin(), arr.end());
        for (int i = 0; i < arr.size() - 1; i++)
        {
            diff = min(diff, arr[i + 1] - arr[i]);
        }
        for (int i = 0; i < arr.size() - 1; i++)
        {
            if ((arr[i + 1] - arr[i]) == diff)
            {
                ans.push_back({arr[i], arr[i + 1]});
            }
        }
        return ans;
    }
};

File: Leetcode-Problems/Minimum Absolute Difference.java
class Solution
{
public
    List<List<Integer>> minimumAbsDifference(int[] arr)
    {
        List<List<Integer>> ans = new ArrayList<>();
        Arrays.sort(arr);
        int mini = Integer.MAX_VALUE;
        for (int i = 1; i < arr.length; i++)
        {
            mini = Math.min(mini, arr[i] - arr[i - 1]);
        }
        for (int i = 1; i < arr.length; i++)
        {
            ArrayList<Integer> temp = new ArrayList<Integer>();
            if (arr[i] - arr[i - 1] == mini)
            {
                temp.add(arr[i - 1]);
                temp.add(arr[i]);
            }
            if (!temp.isEmpty())
                ans.add(temp);
        }
        return ans;
    }
}